framework/include/userobject/ElementUserObject.h

```diff
@@ -63,12 +63,20 @@ class ElementUserObject :
   virtual void execute() = 0;
 
   /**
+   * This function will be called with the shape functions for jvar initialized. It
+   * can be used to compute Jacobian contributions of the UserObject.
+   */
+  virtual void executeJacobian(unsigned int /*jvar*/) {}
+
+  /**
    * Must override.
    *
    * @param uo The UserObject to be joined into _this_ object.  Take the data from the uo object and "add" it into the data for this object.
    */
   virtual void threadJoin(const UserObject & uo) = 0;
 
+  bool requestedJacobian(unsigned int var) const;
```

Comment:
Wait, we already loop over a vector of variables, the ones registered in assembly! We initialize the shape function once for every user object, in the loop over the UOs we check if we really need to call ```executeJacobian```. I think we can just drop this alltogether and call an empty function (or bail out depending on the value of jvar).